controllers/vsphereclusteridentity_controller.go

```diff
@@ -181,9 +189,16 @@ func (r clusterIdentityReconciler) reconcileDelete(ctx _context.Context, identit
 	if ctrlutil.ContainsFinalizer(secret, legacyIdentityFinalizer) {
 		ctrlutil.RemoveFinalizer(secret, legacyIdentityFinalizer)
 	}
+
+	// TODO (killianmuldoon): Consider whether we should have a finalizer on the secret and, if so, whether this finalizer should be renamed.
 	ctrlutil.RemoveFinalizer(secret, infrav1.SecretIdentitySetFinalizer)
```

Comment:
Don't we already have a finalizer on the Secret?

Reply:
Q is whether it should be removed.

Reply:
Missed the "should"